### ISSUE TITLE
Building Docker Alpine:3.7

### ISSUE THREAD

**Issue by MetalBlueberry** [tdlib/td#7]
Hi all,

I'm trying to build the project using Docker with an image based in Alpine3.7.
here is the Dockerfile

```
FROM alpine:3.7 as builder
LABEL Name=td-builer Version=0.0.1

RUN apk update
RUN apk add --update alpine-sdk 
RUN apk add --update openssl-dev
RUN apk add --update cmake
RUN apk add --update gperf
RUN apk add --update zlib-dev

WORKDIR /
COPY ./td /td

WORKDIR /td
RUN mkdir build
WORKDIR build
RUN cmake ..
RUN cmake --build .
RUN cmake --build . --target install
```

I get the following error

```
[  5%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/Stat.cpp.o
/td/tdutils/td/utils/port/Stat.cpp: In function 'td::Stat td::detail::from_native_stat(const stat&)':
/td/tdutils/td/utils/port/Stat.cpp:54:55: error: 'const struct stat' has no member named 'st_mtimensec'; did you mean 'st_mtim'?
   res.mtime_nsec_ = buf.st_mtime * 1000000000ll + buf.st_mtimensec;
                                                       ^~~~~~~~~~~~
/td/tdutils/td/utils/port/Stat.cpp:55:55: error: 'const struct stat' has no member named 'st_atimensec'; did you mean 'st_atim'?
   res.atime_nsec_ = buf.st_atime * 1000000000ll + buf.st_atimensec;
                                                       ^~~~~~~~~~~~
make[2]: *** [tdutils/CMakeFiles/tdutils.dir/build.make:255: tdutils/CMakeFiles/tdutils.dir/td/utils/port/Stat.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:556: tdutils/CMakeFiles/tdutils.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

I've found related issues in other projects
[httpuv/issues/63](https://github.com/rstudio/httpuv/issues/63)
but I can't figure out how to apply it to this one.

Thank you!

**Comment by isopen** [comment]
Try to bypass the mountaineering with this
https://github.com/isopen/docker/blob/master/tdlib/Dockerfile

**Comment by MetalBlueberry** [comment]
Hi @isopen 
That docker image is based on debian:9, thus it compiles without problem.
also, I can compile it using an ubuntu based image, but the idea is to use Alpine.
any other suggestion?

**Comment by isopen** [comment]
I understand that we are not looking for easy ways. We want to minimize the image :)
Then try this:
```
tdlib/td/tdutils/td/utils/port/Stat.cpp54:55
change it so:
func Stat from_native_stat(const struct ::stat &buf)
...
#else
  res.mtime_nsec_ = buf.st_mtime * 1000000000ll + buf.st_mtim.tv_nsec;
  res.atime_nsec_ = buf.st_atime * 1000000000ll + buf.st_atim.tv_nsec;
#endif
...
```

**Comment by levlam** [comment]
As a workaround, you can add || 1 to line Stat.cpp:50:
#if defined(_BSD_SOURCE) || defined(_SVID_SOURCE) || _POSIX_C_SOURCE >= 200809L || _XOPEN_SOURCE >= 700 || TD_EMSCRIPTEN

We will rewrite that condition in more robust way.

**Comment by isopen** [comment]
Hi @MetalBlueberry 
Do this in the sys/stat.h structure
st_mtimensec
st_atimensec
https://opensource.apple.com/source/xnu/xnu-201.5/bsd/sys/stat.h.auto.html
Works are not much :)))

**Comment by isopen** [comment]
Try this bike:
https://github.com/tdlib/td/issues/6
On your computer should be -j 2

I think ideally for a comfortable build you need 12 cores and 32GB RAM

**Comment by MetalBlueberry** [comment]
I think that this issue can be closed.

**Comment by isopen** [comment]
You can reopen this report :) @levlam Not yet fixed
https://github.com/isopen/docker/issues/2

**Comment by levlam** [comment]
The issue should be fixed in TDLib 1.2.0.
